fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePlanner.java

```diff
@@ -263,6 +264,15 @@ private LuceneScanParameters getSpecialScan(@Nonnull LucenePlanState state, @Non
         return null;
     }
 
+    @Nonnull
+    private LuceneQueryComponent tryPushResidual(@Nonnull LuceneQueryComponent luceneQueryComponent, @Nonnull final List<String> prefix) {
+        if (prefix.isEmpty()) {
+            return luceneQueryComponent;
+        }
+        final var name = String.join("_", prefix);
+        return new LuceneQueryComponent(luceneQueryComponent.getQuery(), luceneQueryComponent.getFields().stream().map(field -> name + "_" + field).collect(Collectors.toList()));
```

Comment:
This can lose things like `luceneQueryHighlightParameters`, which seems orthogonal to the fix. Perhaps `LuceneScanQueryParameters` needs a `withOtherFields` or something like that to do it maintainably.

Reply:
Fixed.